shub/exceptions.py

```diff
@@ -107,4 +107,4 @@ def custom_showwarning(message, *args, **kwargs):
         warnings.showwarning = custom_showwarning
         warnings.warn(msg, category=category)
     finally:
-        warnings.formatwarning = old_showwarning
+        warnings.showwarning = old_showwarning
```

Comment:
The MemoryError is sorted @vshlapakov  :)

The error was happening because once we're changing `.showwarning` to custom we weren't changing it back, for some reason it was written `.formatwarning` rather than `.showwarning`.

So when pytest were trying to use `showwarning` it was finding something (our custom implementation) and `.formatwarning` wasn't what it is supposed to be anymore :)

Reply:
Oh, nice, good catch :+1: